### PR TITLE
remove mock-data scripts and use catalog.locations for adding examples

### PR DESCRIPTION
**Pull request by Rugvip**
Figured we can clean up all of the extra steps required to add mock data to the catalog, and just use static location configurations in `app-config.yaml` instead. It's also easy to override in `app-config.local.yaml` for local development.

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -76,23 +76,27 @@ catalog:
...
+    - type: github
+      target: https://github.com/spotify/backstage/blob/master/plugins/techdocs-backend/examples/documented-component/documented-component.yaml
+
+    # Backstage example APIs
+    - type: github
+      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/all-apis.yaml
+
+    # Backstage example templates
+    - type: github
+      target: https://github.com/spotify/backstage/blob/master/plugins/scaffolder-backend/sample-templates/all-templates.yaml
```
Comment (freben) [inline app-config.yaml:99]: Do we need rules for these?

File: packages/create-app/templates/default-app/app-config.yaml.hbs [hunk 2]
```diff
@@ -50,26 +50,13 @@ auth:
...
-      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/shuffle-api-component.yaml
-    - type: github
-      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/hello-world-api.yaml
-    - type: github
-      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/streetlights-api.yaml
+    - type: file
+      target: https://github.com/spotify/backstage/blob/master/packages/catalog-model/examples/all-components.yaml
+
+    # Backstage example APIs
+    - type: file
```
Comment (freben) [inline packages/create-app/templates/default-app/app-config.yaml.hbs:57]: type github for these two

**Review by freben — APPROVED** [review]

**Review by OrkoHunter — APPROVED** [review]
It's really amazing to remove the extra step to generate mock data. And there is one example docs component out-of-the box, so that's nice.

+1 from me. :)

Screenshot 2020-09-12 at 14 32 18

**Comment by OrkoHunter** [comment]
Something wrong with the [zombie](https://github.com/assaf/zombie) package for E2E test on Windows. It randomly pops up sometimes. 😅 
I wonder if we should switch to a more stable headless browser for E2E testing - [puppeteer](https://github.com/puppeteer/puppeteer) is a good one.

File: app-config.yaml (hunk 1 above)
Comment (Rugvip) [reply app-config.yaml:99]: Opted for going with an aggregated file, and just have the global rule instead. Rules don't get carried forward in location entities, and it's awkward to implement that atm.

**Comment by Rugvip** [comment]
@OrkoHunter goal with zombie was to have something very lightweight that can just do a very quick and dirty verification. We're running more elaborate e2e tests with Cypress. It's idd only an issue on the Windows runner, so would wanna see if we can work around that first instead of swapping it out. I can also say that my experience with puppeteer is no better than zombie x)

**Comment by OrkoHunter** [comment]
@Rugvip Ah, I see. Thanks for explaining.
(My experience with all headless browsers is the same - try/catch/try/catch until it works. xD)

**Pull request merged**